packages/commons/src/events/utils.ts

```diff
@@ -69,6 +73,18 @@ export function getDeclaration(configuration: EventConfig) {
   return configuration.declaration
 }
 
+export function getCustomActionFields(eventConfig: EventConfig): FieldConfig[] {
+  const customActions = eventConfig.actions.filter(
+    (action): action is CustomActionConfig => action.type === ActionType.CUSTOM
+  ) satisfies CustomActionConfig[]
+
+  if (!customActions.length) {
+    return []
+  }
+
+  return customActions.flatMap((action) => action.form)
+}
```

Comment:
This would currently cause issues if we have fields with the same id in different custom actions.

We should not get the fields for _all_ custom actions, instead just the fields of the custom action we are executing now.

Lets use this function for getting the config of the correct action: https://github.com/opencrvs/opencrvs-core/blob/develop/packages/commons/src/events/utils.ts#L100

Reply:
Done [here](https://github.com/opencrvs/opencrvs-core/pull/11094/commits/09c9de0feadcef37d2597655e935e4ddb7cf3ce0)

Refactored to call custom action configs with improved types. Leave any improvement suggestions if necessary.